### PR TITLE
Add missing no-failed-unlock assertions to integration tests

### PR DESCRIPTION
**Pull request by jomuel**
Fixes: #4369

This is a follow-up for pr #4469. It adds the assertions for no unexpected unlock failures to some tests that are still missing them. With it all tests should be covered; see issue #4369 for the list of intentionally excluded tests.

### CODE REVIEW

File: raiden/tests/integration/transfer/test_refundtransfer.py [hunk 1]
```diff
@@ -102,7 +103,14 @@ def test_refund_messages(raiden_chain, token_addresses, deposit, network_wait):
     # This channel was exhausted to force the refund transfer except for the fees
     with gevent.Timeout(network_wait):
         wait_assert(
-            assert_synced_channel_state, token_network_address, app1, 0, [], app2, deposit * 2, []
+            assert_succeeding_transfer_invariants,
+            token_network_address,
+            app1,
```
Comment (palango) [inline raiden/tests/integration/transfer/test_refundtransfer.py]: Please add kwargs here.

File: raiden/tests/integration/long_running/test_integration_events.py [hunk 2]
```diff
@@ -431,7 +435,7 @@ def test_secret_revealed_on_chain(
         secret=secret,
     )
 
-    with gevent.Timeout(10):
+    with watch_for_unlock_failures(*raiden_chain), gevent.Timeout(10):
```
Comment (palango) [inline raiden/tests/integration/long_running/test_integration_events.py]: Currently `watch_for_unlock_failures` check that no bad event among all events in the raiden DB. There are two possible improvements I see here:
- Wrap the whole test with `watch_for_unlock_failures` and have everything checked
- Change `watch_for_unlock_failures` to store the number of events as the `offset` parameter for each app  during initialization and then use it when getting events.

I'm leaning towards option 2 as it prevents false positives and allows for tests with intentional unlock failures.

File: raiden/tests/integration/long_running/test_stress.py [hunk 3]
```diff
@@ -336,6 +340,7 @@ def assert_channels(
         )
 
 
+@pytest.mark.skip(reason="flaky, see https://github.com/raiden-network/raiden/issues/4803")
```
Comment (palango) [inline raiden/tests/integration/long_running/test_stress.py]: Did you hit this in this PR? It seems like this doesn't happen often.

File: raiden/tests/integration/long_running/test_integration_events.py (hunk 2 above)
Comment (jomuel) [reply raiden/tests/integration/long_running/test_integration_events.py]: I've been wondering about how to improve `watch_for_unlock_failures` too. The offset parameter is a good idea, I will add it.

File: raiden/tests/integration/long_running/test_stress.py (hunk 3 above)
Comment (jomuel) [reply raiden/tests/integration/long_running/test_stress.py]: Yes. It also happened in the build of #5266, and in about half the times I ran the stress test locally on `develop`. Not sure why it should suddenly happen more often though.

File: raiden/tests/utils/events.py [hunk 4]
```diff
@@ -131,18 +131,20 @@ def must_have_events(event_list: List[TM], *args) -> bool:
...
 
-def has_unlock_failure(raiden: RaidenService) -> bool:
-    events = raiden.wal.storage.get_events()  # type: ignore
-    return has_event_of_types(events, (EventUnlockFailed, EventUnlockClaimFailed))
+def count_unlock_failures(events: List[Event]) -> int:
+    return count_event_of_types(events, (EventUnlockClaimFailed, EventUnlockFailed))
+
+
+def has_unlock_failure(raiden: RaidenService, offset: int = 0) -> bool:
+    return count_unlock_failures(raiden.wal.storage.get_events()) > offset  # type: ignore
```
Comment (palango) [inline raiden/tests/utils/events.py]: Why is the `type: ignore` needed here?

Comment (jomuel) [reply raiden/tests/utils/events.py]: Otherwise we would have to type-assert that `raiden.wal` is not `None`.

**Pull request merged**